### PR TITLE
Make timeouts longer and more consistent

### PR DESCRIPTION
**Pull request by andy-sweet**
# References and relevant issues
Following up on the comment in #6187

# Description
This generally increases the timeouts in the prerelease and pull request workflows because they have been causing spurious cancellations when runs are almost complete. The new times are mostly consistent with those in `test_comprehensive.yml`, but I also made an effort to use a smaller set of values to avoid becoming too dependent on particular measurements at particular times.

### CODE REVIEW

**Comment by andy-sweet** [comment]
@Carreau : marked this as draft, because I'm not confident about my reasoning here (in the PR description). Generally, I think timeouts should be generous and are mostly there to stop runs that will not stop.

It would be nice to set the default once from 600 to 60 minutes, but I'm not sure that's a feature yet though it has [been discussed](https://github.com/orgs/community/discussions/10690).

**Comment by andy-sweet** [comment]
Also, I don't have evidence to back it up, but my qualitative impression is that `test_qt_dims_2.py` is mostly the test module causing problems (running indefinitely) - maybe we could just disable/remove those tests temporarily either in this PR or another.

**Comment by Carreau** [comment]
Up to you, my timeout when introduced were already pretty conservative, so it might be soem things are now taking much longer. 

One of the reason to not make them much larger than the actual time, is that if a item blocks, it is going to occupy a worker for the full time of the timeout. So for example 10->60 seem a bit too much. If examples run generally in say 5 minutes, and the timeout is 60, if they haven't completed in 10, they are not going to complete in 60, and holding a worker for another  50 minutes, potentially delaying other PRs... so I'm not too sure of making it 60 minutes everywhere.

**Comment by Carreau** [comment]
>> We should be uploading pytest json report, maybe we can make it an unconditional upload even on cancell / failure (it it possible?) that should let us extract which one are regularly failing after the facts ?

> That sounds useful. Though I guess we can't easily distinguish between timeout due to a hanging/stuck test vs. timeout due to impatience. 

Let me rephrase.

_if_ report get uploaded/generated even with timeout, we could have a record of which test was running when it was cancelled. If the same test comes back again and again, then it's likely a test that is regularly stuck.

Anyway I don't think pytest report generate the report if it gets interrupted.

**Comment by andy-sweet** [comment]
Looks like this might be superseded by https://github.com/napari/napari/pull/6156 and I've seen fewer/no timeouts recently, so I'm going to close this. Would still suggest being cautious about making timeouts too tight in the future to leave room for CI having slow days and for future additions to tests.

**Comment by andy-sweet** [comment]
I definitely tempted fate and paid for it :D

https://github.com/napari/napari/actions/runs/6228462545/job/16905251930?pr=6230#step:8:287

**Comment by Czaki** [comment]
In #6156 you may see added time print information. It is added to allow understanding where is the problem. If the timeout is too short or the tests just hangs.
